### DISCUSSION TITLE
Missing standard orientation

### DISCUSSION THREAD

**Discussion by NovakEvana** [mtex-toolbox/mtex#1640] [Ask Anything] [answered]
Hello everyone, 

The image below was generated based on a template code. However, I don't see the cube component displays at 4 corners when plotting them at phi2 = 0. Could any help me understand why the cube orientation is missing here? The code is attached below. Thanks,

![image](https://user-images.githubusercontent.com/117882416/200991192-6029cd71-4580-4530-a220-9b048faae7eb.png)

```
cs = crystalSymmetry('m-3m');
ss = specimenSymmetry('orthorhombic');
components = [...
  orientation.goss(cs,ss),...
  orientation.brass(cs,ss),...
  orientation.cube(cs,ss),...
  orientation.cubeND22(cs,ss),...
  orientation.cubeND45(cs,ss),...
  orientation.cubeRD(cs,ss),...
  orientation.copper(cs,ss),...
  orientation.PLage(cs,ss),...
  orientation.QLage(cs,ss),...
  ];
close all
for i = 1:length(components)
  plotSection(components(i), 'add2all', 'MarkerColor', ind2color(i),...
    'DisplayName', round2Miller(components(i),'LaTex'))
end

legend('show','interpreter','LaTeX','location','southeast','FontSize',1.2*getMTEXpref('FontSize'));
```

**Comment by kilir** [comment, marked as answer]
Hi,
indeed this should not be the case but doing   `plotSection(components(i), 'add2all', 'preserveOrder')` should help.
Cheers,
Rüdiger

**Reply by NovakEvana** [reply]
Hi Rudiger, can you explain why it should not be the case?

**Reply by kilir** [reply]
Well, because as you noted, the points for the cube orientation should show up when being plotted. Did '`preserveOrder'` work for you?

**Reply by NovakEvana** [reply]
Yes, it did. Thank you so much for your help. Could you elaborate a little bit why it didn't show up at the first place?

**Comment by NovakEvana** [comment]
Hi @kilir, 

It seems like the cube component is still not reflecting correctly even though I've added ```'preserveOrder'``` as your advice. The missing cubes are at 2 rightmost corners at phi2=0 and at the top horizontal position at phi2 = 45. I've added the standard components so that you can refer to. My code is as follow:

```
cs = crystalSymmetry('m-3m');
ss = specimenSymmetry('triclinic');
figure;
plotSection(orientation.cube(cs,ss), 'phi2',[0 45]*degree,'add2all',...
    'preserveOrder', 'MarkerColor', 'black','Marker','s')
hold on;
```

### **My results:**
![Screenshot 2022-11-13 005850](https://user-images.githubusercontent.com/117882416/201509920-ec8658c7-4a24-405c-9918-db154e069062.jpg)


### **Reference standard:**
![image](https://user-images.githubusercontent.com/117882416/201509928-12f6d678-66a5-4703-93da-bb1a644da6e8.png)

**Reply by kilir** [reply]
Can you try this 105da601150f545517521d4243dbf61dc00430e0 ?
It should fix the non-appearing points without using 'preserveOrder'.

**Reply by NovakEvana** [reply]
Hi @kilir , 

Thanks for your answer. The `'preserveOrder'` is fixed but the cube points as stated above are still missing.

Edit: I've created a [new thread](https://github.com/mtex-toolbox/mtex/discussions/1653) for this issue. Thanks.

**Reply by kilir** [reply]
Hi,

two things: In order to "convince" mtex to draw two identical points (those on the right side of your 0 degree section) you can symmetrise your orientations before using `plotSection()`.  We might discuss whether it is a bug that identical points do not appear twice.

However, could you explain why there should be points in the 45 degree section?

Cheers,
Rüdiger

**Reply by NovakEvana** [reply]
Hi @kilir ,

`However, could you explain why there should be points in the 45 degree section?`
Because my crystal symmetry is cubic so there would be repetition between Euler (0 0 0) and (45 0 45). Thanks.

EDIT: I think the `symmetrise `doesn't work either.
